marklogic-data-hub-central/ui/src/components/entities/matching/matching-card.test.tsx

```diff
@@ -116,4 +119,161 @@ describe("Matching cards view component", () => {
+    expect(getByText("matchCustomersEmpty")).toBeInTheDocument();
+    userEvent.click(getByTestId("matchCustomers-run"));
+    wait(() => {
+      expect(getByText("Run Step in a new flow")).toBeInTheDocument();
+      userEvent.click(getByText("Run Step in a new flow"));
+    });
+    wait(() => {
+      expect(screen.getByText("NewFlow")).toBeInTheDocument();
+      userEvent.type(screen.getByPlaceholderText("Enter name"), "testFlow");
+      userEvent.click(screen.getByText("Save"));
+    });
+    // Check if the /tiles/run/add route has been called
+    wait(() => { expect(mockHistoryPush).toHaveBeenCalledWith("/tiles/run/add"); });
+  });
+
+  it("can add a step to an existing flow", () => {
+    const {getByText, getByTestId, getByLabelText, queryByText} =  render(
+      <Router>
+        <MatchingCard {...defaultProps}/>
+      </Router>
+    );
+
+    expect(getByText("matchCustomers")).toBeInTheDocument();
+
+    //mouseover to trigger flow menu
+    fireEvent.mouseOver(getByTestId("Customer-matchCustomers-step"));
+    wait(() => {
+      expect(getByText("Run Step in an existing flow")).toBeInTheDocument();
+      //mouse out for card view
+      fireEvent.mouseOut(getByTestId("Customer-matchCustomers-step"));
+    });
+
+    wait(() => {
+      expect(queryByText("Run Step in an existing flow")).toBeNull();
+      fireEvent.mouseOver(getByTestId("Customer-matchCustomers-step"));
+    });
+
+    wait(() => {
+      expect(getByText("Run Step in an existing flow")).toBeInTheDocument();
+      userEvent.click(getByTestId("matchCustomers-flowsList"));
+      userEvent.click(getByLabelText("customerJSONFlow-option"));
+    });
+
+    wait(() => {
+      expect(getByLabelText("step-not-in-flow-run")).toBeInTheDocument();
+      userEvent.click(getByTestId("matchCustomers-to-test-Confirm"));
+    });
+    // Check if the /tiles/run/add route has been called
+    wait(() => { expect(mockHistoryPush).toHaveBeenCalledWith("/tiles/run/add"); });
+  });
+
+  it("can add a step to an existing flow from run menu", () => {
+    const {getByText, getByTestId, getByLabelText} =  render(
+      <Router>
+        <MatchingCard {...defaultProps}/>
+      </Router>
+    );
+
+    expect(getByText("matchCustomers")).toBeInTheDocument();
+    expect(getByText("matchCustomersEmpty")).toBeInTheDocument();
+    userEvent.click(getByTestId("matchCustomers-run"));
+    wait(() => {
+      expect(getByText("Run Step in an existing flow")).toBeInTheDocument();
+      userEvent.click(getByTestId("matchCustomers-run-flowsList"));
+      userEvent.click(getByLabelText("customerJSONFlow-run-option"));
+    });
+    wait(() => {
+      expect(getByLabelText("step-not-in-flow-run")).toBeInTheDocument();
+      userEvent.click(getByTestId("matchCustomers-to-test-Confirm"));
+    });
+    // Check if the /tiles/run/add route has been called
+    wait(() => { expect(mockHistoryPush).toHaveBeenCalledWith("/tiles/run/add"); });
```

Comment:
@brucean52 can we also add validations for checking whether we are getting redirected to flows and the match step is added to the existing flow/the newly added flow.

Reply:
As per discussion with Bruce, will be taken care as part of DHFPROD-4548.

Reply:
As we discussed, there's already a RTL test for this and I will add additional e2e testing for this in a later PR.